### PR TITLE
[konflux migration] Add Konflux Pipelines configs

### PR DESCRIPTION
**Pull request by esron**
[APPSRE-11016](https://issues.redhat.com/browse/APPSRE-11016)

Adding Konflux configuration.

Updating the main Dockerfile to use muilt stage builds so we can use the same Dockerfile to run tests and build the production artifact.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
 FROM registry.access.redhat.com/ubi9/go-toolset:1.21.13-2.1729776560 as builder
 WORKDIR /build
 COPY --chown=1001:0 . .
-RUN make gobuild
+RUN make gobuild golint gotest
```
Comment (hemslo) [inline Dockerfile:4]: `gobuild` will call `gotest`, we should reimplement checks in https://github.com/app-sre/go-qontract-reconcile/blob/main/pr_check.sh by merge

Comment (esron) [reply Dockerfile:4]: Got it.

Comment (hemslo) [reply Dockerfile]: the test structure is a little compex in this repo, `make update-schema` is to update `schema.graphql` file, which requires integration test like setup. `make validate-schema` just build `Dockerfile.validate` which is almost identical to `Dockerfile`. We can remove that file, by replacing usages at https://github.com/app-sre/go-qontract-reconcile/blob/6965aea62d84c37473740ee9f608828b3c7a9bc6/Makefile#L41-L43 to use `Dockerfile` and target `builder`.

**Review by hemslo — APPROVED** [review]
LGTM!

**Comment by esron** [comment]
I needed to push a dummy commit so the pipeline runs with the recently added task.

**Pull request merged**